onnxruntime/core/providers/coreml/builders/impl/binary_op_builder.cc

```diff
@@ -54,6 +54,20 @@ bool CheckIfBothInputShapesMatch(const Node& node, const logging::Logger& logger
                     y_shape_proto->dim().begin(), y_shape_proto->dim().end(),
                     dim_eq);
 }
+
+bool ShouldUseFloorDiv(const Node& node, const logging::Logger& logger) {
+  const auto& input0 = *node.InputDefs()[0];
+  const auto& input1 = *node.InputDefs()[1];
+  int32_t input_type0 = ONNX_NAMESPACE::TensorProto_DataType_UNDEFINED;
+  int32_t input_type1 = ONNX_NAMESPACE::TensorProto_DataType_UNDEFINED;
+  GetType(input0, input_type0, logger);
+  GetType(input1, input_type1, logger);
+
+  return input_type0 == ONNX_NAMESPACE::TensorProto_DataType_INT32 ||
+         input_type0 == ONNX_NAMESPACE::TensorProto_DataType_INT64 ||
+         input_type1 == ONNX_NAMESPACE::TensorProto_DataType_INT32 ||
+         input_type1 == ONNX_NAMESPACE::TensorProto_DataType_INT64;
```

Comment:
both inputs in ONNX have to be the same type so you don't need to check input_type0 and input_type1